Modules/Module_Processing/Module_ASL/xASL_wrp_ProcessM0.m

```diff
@@ -120,10 +131,40 @@ function xASL_wrp_ProcessM0(x)
 
 			% Find the smallest of the previously selected positive nonzero TEs
 			minTE = min(uniqueNonzeroTE);
-			fprintf('Multi-TE M0 present. Using the shortest TE=%.1f ms for M0 calibration.\n', minTE*1000);
+			% JSON values were converted from BIDS (seconds) to Legacy (millisecond) during JSON reading
+			fprintf('Multi-TE M0 present. Using the shortest TE = %.1f ms for M0 calibration.\n', minTE);
+
+			% Find the correct indices and update the jsonM0 parameter
+			indMinTE = jsonM0.Q.EchoTime == minTE;
+			jsonM0.Q.EchoTime = jsonM0.Q.EchoTime(indMinTE);
+			if isfield(jsonM0, 'RepetitionTime') && length(jsonM0.RepetitionTime) > 1
+				jsonM0.RepetitionTime = jsonM0.RepetitionTime(indMinTE);
+			end
+
+			% Select the minimal TEs
+			imM0 = imM0(:, :, :, indMinTE);

```

Comment:
I would rewrite ind to index for readability but not a big deal of course

Reply:
Yes. agree. Done